### PR TITLE
131 input such as `''`, `""` displays unappropriate err msg

### LINKED ISSUE #131: Input such as `''`, `""` displays unappropriate err msg
```bash
minishell$ ''
minishell: /Users/yliu/.pyenv/shims/: Is a directory
minishell$ ""
minishell: /Users/yliu/.pyenv/shims/: Is a directory
```

### PR DESCRIPTION
**Pull request by rask24**
fix #131

- **Fix condition of search_for_command**
- **Add e2e tests**

## Summary by CodeRabbit

- **New Features**
	- Enhanced command search functionality to ensure only valid and executable command paths are returned.
	- Restructured main execution flow for improved clarity and input handling.

- **Bug Fixes**
	- Improved error handling in shell command execution.

- **Tests**
	- Added tests to verify correct error handling for empty commands and the `..` command.

### CODE REVIEW

**Comment by rask24** [comment]
## Examples

``` bash
minishell$ ''
minishell: : command not found
minisehll$ ""
minishell: : command not found
minishell$: .
minishell: .: command not found
minishell$: ..
minishell: ..: command not found
```

File: src/main.c [hunk 1]
```diff
@@ -34,30 +34,40 @@ static void	destroy_ctx(t_ctx *ctx)
...
 			break ;
 		}
+		if (ft_strcmp(input, "") == 0)
+		{
+			free(input);
+			continue ;
+		}
 		exec(input, ctx);
 		add_history(input);
 		free(input);
```
Comment (alex216) [inline src/main.c:56]:
```suggestion
		free(input);
		restore_terminal_configuration(original_termios);
```

**Comment by alex216** [comment]
@rask24 please check the comment!

File: src/main.c (hunk 1 above)
Comment (rask24) [reply src/main.c:56]: Exactly 👍🏼

**Review by alex216 — APPROVED** [review]
@rask24 ok!

**Pull request merged**